2024-03-17 17:18:44.068938: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.070222: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.071777: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.072326: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.102276: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.102244: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.103312: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.103560: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.112094: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.112558: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.113767: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.113767: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.120998: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.121187: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.121553: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:44.121508: W external/xla/xla/service/hlo_rematerialization.cc:2941] Can't reduce memory use below -40.08GiB (-43041683865 bytes) by rematerialization; only reduced to 196.00GiB (210453397520 bytes), down from 196.00GiB (210453397520 bytes) originally
2024-03-17 17:18:54.145010: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_0_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.145024: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_1_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.145176: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
2024-03-17 17:18:54.145176: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
2024-03-17 17:18:54.145153: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_2_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
E0317 17:18:54.145256 3651558 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


E0317 17:18:54.145258 3651557 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.145394: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.145538 3651559 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.146462: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_3_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.146593: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.146675 3651560 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.177318: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_2_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.177537: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.177661 2447181 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.178920: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_3_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.179019: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.179078 2447182 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.180518: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_1_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.180631: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
2024-03-17 17:18:54.180694: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_0_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
E0317 17:18:54.180707 2447180 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.180801: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.180871 2447179 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.189088: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_1_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.189331: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.189451 3200464 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.189697: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_0_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.189794: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.189854 3200463 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.190498: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_2_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.190431: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_3_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.190563: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
2024-03-17 17:18:54.190595: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.190628 3200466 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


E0317 17:18:54.190655 3200465 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.202417: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_1_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.202527: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.202588 1742500 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.202613: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_2_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.202602: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_3_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.202733: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
2024-03-17 17:18:54.202725: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.202803 1742502 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


E0317 17:18:54.202805 1742501 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


2024-03-17 17:18:54.204085: W external/tsl/tsl/framework/bfc_allocator.cc:482] Allocator (GPU_0_bfc) ran out of memory trying to allocate 64.00GiB (rounded to 68719476736)requested by op 
2024-03-17 17:18:54.204183: W external/tsl/tsl/framework/bfc_allocator.cc:494] ________*********___________________________________________________________________________________
E0317 17:18:54.204241 1742499 pjrt_stream_executor_client.cc:2804] Execution of replica 0 failed: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 107, in <module>
    run_benchmark(global_shape,nb_nodes, output_path)
  File "/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py", line 54, in run_benchmark
    do_fft(global_array).block_until_ready()
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 68719476736 bytes.
BufferAssignment OOM Debugging.
BufferAssignment stats:
             parameter allocation:    2.00GiB
              constant allocation:         0B
        maybe_live_out allocation:   64.00GiB
     preallocated temp allocation:   64.00GiB
  preallocated temp fragmentation:         0B (0.00%)
                 total allocation:  130.00GiB
              total fragmentation:       128B (0.00%)
Peak buffers:
	Buffer 1:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 2:
		Size: 64.00GiB
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: fft
		Shape: c64[2048,2048,2048]
		==========================

	Buffer 3:
		Size: 2.00GiB
		Entry Parameter Subshape: f32[128,2048,2048]
		==========================

	Buffer 4:
		Size: 16B
		Operator: op_name="jit(do_fft)/jit(main)/jit(fft)/fft[fft_type=FftType.FFT fft_lengths=(2048, 2048, 2048)]" source_file="/gpfsdswork/projects/rech/tkc/commun/jaxDecomp/benchmarks/jaxfft.py" source_line=41
		XLA Label: all-gather-start
		Shape: (c64[128,2048,2048], c64[2048,2048,2048])
		==========================


srun: error: r6i6n8: tasks 5-7: Exited with exit code 1
srun: Terminating StepId=1125360.3
srun: error: r7i2n6: tasks 8-9,11: Exited with exit code 1
srun: error: r7i5n3: tasks 13-15: Exited with exit code 1
srun: error: r6i6n0: tasks 1-3: Exited with exit code 1
slurmstepd: error: *** STEP 1125360.3 ON r6i6n0 CANCELLED AT 2024-03-17T17:18:54 ***
srun: error: r6i6n8: task 4: Exited with exit code 1
srun: error: r7i5n3: task 12: Exited with exit code 1
srun: error: r7i2n6: task 10: Exited with exit code 1
srun: error: r6i6n0: task 0: Exited with exit code 1
